### PR TITLE
Remove usages of findAll for types supported by server-side pagination

### PR DESCRIPTION
**Pull request by richard-cox**
~Blocked on https://github.com/rancher/dashboard/pull/10786~

### Summary
Contributes to https://github.com/rancher/dashboard/issues/9964

### Occurred changes and/or fixed issues
- Where findAll is used try to instead
  - A) fetch a smaller set of resources via existing API methods (like filtering on namespace) 
  - or
  - B) Performance Setting on - Fetch via new API filtering. Setting off - continue as is
- This code wise is based on top of #10786 which does a similar thing but for findAlls rather than LabelSelects

In Addition
- With the steve cache enabled the Workflow list should not be navigatable. This will be replaced at some point with an overview page 
- Fixed `Rancher Backups` --> `Backups`, `Resource` --> `Create`/edit page bottom bar (was not stuck to bottom)

### Technical notes summary
- There's two temp bools in shell/utils/pagination-utils.ts, TEMP_VAI_CACHE_MERGED and TEMP_PERF_ENABLED. These will be removed pre 2.9.0 when the steve cache is merged and we can just run on the FF and perf setting to enable/disable the features
- For this PR would recommend
  - Provision Rancher with the vai image (available on request)
  - Enable the steve cache `Feature Flag`
  - Enable pagination via the `Performance` setting

### Areas or cases that should be tested
- PR wise with the new vai cache image and everything enabled. In addition maybe a bit with it disabled. QA should be looking at more thorough tests as per below

### Areas which could experience regressions
With both the performance pagination setting and feature flag off

- Harvester Machine Config gets correct config maps. `Cluster Management` --> `Cluster` list --> `Create` --> `RKE2` `Harvester` --> `User Data Template` and `Network Data Template` contains correct config maps
- New ResourceLabeledSelect` continues to work. Nav to `local` cluster --> `Cluster Tools` --> `Rancher Backup` `Install` --> Next --> `Use an SE-compatible object store` --> `Credential Secret should show values
- SelectOrCreateAuthSecret continues to work. Nav to `local` cluster --> `Apps` --> `Repositories` --> `Rancher` row `Edit Config` --> `http(s) URL to an index generaed by Helm` --> `Authentication` user can create either type and also select an existing one
- Cloud Credential API Key field correctly populates. (cloud creds should exist, type shouldn't matter ) `Cluster management` --> `Cloud Credential` --> `API Key` column should be populated
- Ingress target should load correctly. Nav to `local` cluster --> `Service Discovery` --> `Ingresses` --> The `Target` columns should contain <url> / <something>
- Ingress workflow detail page shows Certificate link. Nav to `local` cluster --> `Service Discovery` --> `Ingress` --> click on `rancher` --> `Rules` tab --> confirm `Certificate` column is populated
- Ingress create/edit Certifcate secret selection works correct. Nav to `local` cluster --> `Service Discovery` --> `Ingresses` --> `Create` --> `Certificates` tab --> `Certificate - Secret Name` contain at least one option `Default Ingress Controller Certificate` and changing the namespace to `kube-system` should then reveal another option (depends on kube flavour, on k3s its k2s-serving).
- Node Pods list is populated. Nav to `local` cluster --> `Cluster` --> `Nodes` --> click on any node --> `Pods` tab --> Confirm the pods list is populated
- Backup Encryption key can be selected. Nav to `local` cluster --> (install `Rancher Backups`) --> `Rancher Backups` --> `Backups` --> `Create` --> `Encryption` --> Select `Encrypt backsps using... ` --> Drop down below should be populated (conditions for secrets in tooltip must be met)
- Restore Credential secret can be selected. Nav to `local` cluster --> (install `Rancher Backups`) --> `Rancher Backups` --> `Restore` --> `Create` --> Select `An S3-compatible bject store` --> Drop down below should be populated
- ServiceAccounts secret can be selected. Nav to `local` cluster --> (ensure a secret of type `kubernetes.io/dockerconfigjson` has been created in a specific namespace) --> `More Resources` --> `ServiceAccounts` --> `Create` --> select specific namespace --> `Service Account` tab --> `Pull Secrets` should contain secret. Changing the namespace should result in the correct secrets relative to that namespace
- Project Montioring CRUD is not broken. Nav to `local` cluster --> (Install monitoring and prometheus federator apps) --> `Monitoring` --> `Project Monitoring` --> can succesfully create and view one

The above should be repeated with the performance pagination setting and feature flag on. In Addition
- With the steve cache enabled the Workflow list should not be navigatable. This will be replaced at some point with an overview page 

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/edit/networking.k8s.io.ingress/index.vue [hunk 1]
```diff
@@ -45,23 +47,37 @@ export default {
...
-      secrets:               this.$store.dispatch('cluster/findAll', { type: SECRET }),
+
+    const promises = {
       services:              this.$store.dispatch('cluster/findAll', { type: SERVICE }),
       ingressClasses:        this.ingressClassSchema ? this.$store.dispatch('cluster/findAll', { type: INGRESS_CLASS }) : Promise.resolve([]),
       ingressResourceFields: this.schema.fetchResourceFields(),
-    });
+    };
+
+    this.filterByApi = paginationUtils.isEnabled({ rootGetters: this.$store.getters }, { store: 'cluster', resource: { id: SECRET } });
```
Comment (richard-cox) [inline shell/edit/networking.k8s.io.ingress/index.vue:60]: getter dashboard-store `isPaginationEnabled(SECRET)`

**Comment by richard-cox** [comment]
@codyrancher @nwmac I'm having fun getting this one passed the gates (some might be geniune, but a lot of other prs are currently failing). The code should hopefully remain unchanged, would it be possible to review now to save time?

**Review by codyrancher — APPROVED** [review]
I didn't verify Harvester or S3 due to access.

From what I could verify it looks like we're still populating resources just fine.

Do you have a script or something for generating the resources in the future? It's fairly time consuming testing something like this.

**Comment by richard-cox** [comment]
@codyrancher Yeah there's a few requirements on certain types of secrets. I'm aiming to have some e2e tests where we can upload resources via YAML to populate pages.

**Pull request merged**